drivers/ad717x/ad717x.c

```diff
@@ -294,6 +297,58 @@ int32_t AD717X_ReadData(ad717x_dev *device,
         return ret;
 }
 
+/***************************************************************************//**
+* @brief Computes data register read size to account for bit number and status
+* 		 read.
+*
+* @param device - The handler of the instance of the driver.
+*
+* @return 0in case of success or negative code in case of failure.
+*******************************************************************************/
+int32_t AD717X_ComputeDataregSize(ad717x_dev *device)
+{
+	int32_t ret = 0;
+	ad717x_st_reg *reg_ptr;
+	ad717x_st_reg *datareg_ptr;
+	uint16_t case_var;
+
+	/* Get interface mode register pointer */
+	reg_ptr = AD717X_GetReg(device, AD717X_IFMODE_REG);
+	/* Get data register pointer */
+	datareg_ptr = AD717X_GetReg(device, AD717X_DATA_REG);
+
+	case_var = reg_ptr->value & (AD717X_IFMODE_REG_DATA_STAT |
+			AD717X_IFMODE_REG_DATA_WL16);
+
+	/* Compute data register size */
+    switch(case_var) {
+    case 0:
+    	datareg_ptr->size = 3;
+    	break;
+    case AD717X_IFMODE_REG_DATA_WL16:
+    	datareg_ptr->size = 2;
+    	break;
+    case AD717X_IFMODE_REG_DATA_STAT:
+    	datareg_ptr->size = 4;
+    	break;
+    case (AD717X_IFMODE_REG_DATA_STAT | AD717X_IFMODE_REG_DATA_WL16):
+    	datareg_ptr->size = 3;
+    	break;
+    default:
+    	datareg_ptr->size = 3;
+    	break;
+    }
```

Comment:
this can be simplified:
```
datareg_ptr->size = 3; // default
if (case_var == AD717X_IFMODE_REG_DATA_WL16)
        datareg_ptr->size--;
else if (case_var == AD717X_IFMODE_REG_DATA_STAT)
         datareg_ptr->size--;
```
or something like this